cunumeric/deferred.py

```diff
@@ -633,30 +633,19 @@ def get_item(self, key):
             store = rhs.base
             if copy_needed:
                 # Create a new array to be the result
-                result = self.runtime.create_empty_thunk(
-                    index_array.base.shape,
+                result_store = self.context.create_store(
                     self.dtype,
-                    inputs=[self],
+                    shape=index_array.shape,
+                    optimize_scalar=False,
+                )
```

Comment:
@magnatelee Is it OK to keep `optimize_scalar=False` for both future and non-future cases?

Reply:
we want to use scalars wherever applicable, so I don't want us to pass `optimize_scalar=False` blindly